Demo/Demo/SwiftUIComponents/PayPalWebPayments/PayPalWebTransactionView.swift

```diff
@@ -0,0 +1,44 @@
+import SwiftUI
+
+struct PayPalWebTransactionView: View {
+
+    @ObservedObject var payPalWebViewModel: PayPalWebViewModel
+
+    var body: some View {
+        ScrollView {
+            ScrollViewReader { scrollView in
+                VStack {
+                    PayPalWebStatusView(status: .approved, payPalWebViewModel: payPalWebViewModel)
+                    ZStack {
+                        Button("\(payPalWebViewModel.intent.rawValue.capitalized) Order") {
+                            Task {
+                                do {
+                                    try await payPalWebViewModel.completeTransaction()
+                                } catch {
+                                    print("Error capturing order: \(error.localizedDescription)")
+                                }
+                            }
+                        }
+                        .buttonStyle(RoundedBlueButtonStyle())
+                        .padding()
+
+                        if payPalWebViewModel.state == .loading {
+                            CircularProgressView()
+                        }
+                    }
+
+                    if payPalWebViewModel.state == .success && payPalWebViewModel.order?.status == "COMPLETED" {
+                        PayPalWebResultView(payPalWebViewModel: payPalWebViewModel, status: .completed)
```

Comment:
I am thinking of cases where order status COMPLETED is not the end state in some of the vault flows.
How we would track captured/authorized state.

Reply:
According to PPaaS "COMPLETED" is always the result of a successful authorize/capture attempt which is what is being displayed here - the result of a completed call to auth/capture

Reply:
So as we were discussing, in some vaulting flows, it was helpful to surface unexpected "APPROVED" state after authorize/capture. It seems a departure from checking for presence of order after order creation, checking for checkoutResult for approval and here checking for value of order status being "COMPLETED".

Reply:
I think in those cases we can do `viewModel.order?.status == "COMPLETED" || viewModel.order?.status == "APPROVED"`. From the testing I've done thus far we always get `COMPLETED` back from PayPal Web. We won't want to display the result of a successfully authorized/captured transaction for the other status' returned from PPaaS: saved, approved, pending approval, voided, completed, payer action required, and partially completed.

This conditional is specifically for displaying the result of a successful authorize or capture API call, which we only want to display with specific status'.

Reply:
> but you know Jax, that situation made me sus. It was not a few in that situation.

I am not totally sure I follow, but were there cases where the call to authorize/capture returned one of: saved, pending approval, voided, payer action required, or partially completed but was actually successful? If so that doesn't match the description of those status on the authorize/capture API call according to PPaaS, those status all would indicate the transaction was not completed which we would not display the final completed status view for.

> Success case should be some value but we got something else.

The success state is set by us as a result of the `completeOrder` API call. We are checking for both a success state as well as the `order.status` raw value string from the API.

> And in that case we need to surface that.

If we want to surface the other status that should _not_ be in the conditional here. Instead we would want a separate conditional for the other status results. Just want to make sure I am understand what you are saying correctly though?

> This was my original reason for having separate variables of states for each stage of API results, to be able to track them in the vm.

We can still track these! The previous state variables were all being set to a generic order which has now been combined into a singular order. The logic is the exact same (result of each call in the view model updates the order) so we have the exact same information we did previously!

Reply:
yes it was success and returned Authorized after authorize/capture. 
So I think previously, we could track loaded(order) for authorized/capture variable.
I don't really care as long as we can surface all the expected, unexpected results Jax.